articles/views.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any
+
+from rest_framework import generics, status
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework.renderers import JSONRenderer
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from articles.models import Article
+from articles.permissions import IsAuthorEditorOrReadOnly
+from articles.serializers import ArticleSerializer
+
+
+class ArticleListView(generics.ListCreateAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = ArticleSerializer
+    queryset = Article.objects.all()
+    renderer_classes = (JSONRenderer,)
+
+
+class ArticleDetailView(generics.RetrieveUpdateDestroyAPIView):
```

Comment:
What if the slug changes? will it still return the same article in the case for example I shared with you a link that had the old slug?, put a test for that

Reply:
ok